### PR TITLE
fix: enabling gzip was causing eof errors upon decompression

### PR DESCRIPTION
**Pull request by leorinat**
**Description:** 
Fixed issue when enabling compression: gzip 

**Testing:** 
Locally decompressed the generated payloads before and after the fix

### CODE REVIEW

**Review by MovieStoreGuy — APPROVED** [review]

**Comment by MovieStoreGuy** [comment]
Fixes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/30108

**Comment by MovieStoreGuy** [comment]
Hey @leorinat , 

You'll need to add a changelog entry for this PR to be merged, I believe it should be `make chlog-new` and new file will appear for you to fill in.

**Comment by leorinat** [comment]
> Hey @leorinat ,
> 
> You'll need to add a changelog entry for this PR to be merged, I believe it should be `make chlog-new` and new file will appear for you to fill in.

Done, thanks. 
Any way I can trigger the workflow again? I doesn't look like it triggered by just pushing that change

File: exporter/awskinesisexporter/internal/compress/noop_compression.go [hunk 1]
```diff
@@ -26,3 +26,5 @@ func (n noop) Write(p []byte) (int, error) {
 func (n noop) Flush() error {
 	return nil
 }
+
+func (n *noop) Close() error { return n.Close() }
```
Comment (MovieStoreGuy) [inline exporter/awskinesisexporter/internal/compress/noop_compression.go:30]: Aa, this should return nil here I believe.

Comment (leorinat) [reply exporter/awskinesisexporter/internal/compress/noop_compression.go]: great call, thanks! 
fixed it and pushed

**Comment by MovieStoreGuy** [comment]
Its not exactly clear to me why the correctness tests are failing, let check if it is also impacting main.

**Review by jamesmoessis — APPROVED** [review]

**Comment by jordankay13** [comment]
> Fixes #30108

This PR doesn't address the parallel compression bug you referenced here.

**Pull request merged**